academic-services/knowledge-exploration-service/concepts-queries.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Natural Language Queries
+description: 
+ms.topic: reference
+ms.date: 2020-02-07
+---
+
+# How MAKES generates semantic interpretations of natural language queries
+
+This article describes how MAKES generates semantic interpretations of a natural language query that enable entities to be retrieved. It is intended for developers using MAKES who require a deeper understanding of the architecture and techniques used.
+
+## Architecture overview and diagram
+
+Processing a natural language query is a multi-stage process that starts with lexical analysis of the query, allowing the query to be rewritten for optimal search performance. The rewritten query is then parsed and interpretations hypothesized using a [Context Sensitive Grammar (CSG)](https://en.wikipedia.org/wiki/Context-sensitive_grammar). Valid hypotheses are resolved into full semantic query expressions which are in turn used to generate the complete interpretation response.
+
+:::image type="content" source="media/concept-semantic-interpretation-flow.png" alt-text="MAKES Semantic Interpretation Generation Architecture":::
+
+## Stage 1: Lexical analysis
+
+:::image type="content" source="media/concept-semantic-interpretation-lexical.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Lexical Analysis":::
+
+Lexical analysis of the user query is done to help ensure that valid interpretation hypotheses can be generated.
+
+Currently this is composed of normalization and query operator extraction.
+
+### Normalization
+
+Normalization transforms the query based on specific rules:
+
+* Removing non-essential words (stopwords, such as "the", "and", etc.)
```

Comment:
MAKES will be doing this very soon, and conceptually this is where this would happen

---
In reply to: [377172605](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/108#discussion_r377172605) [](ancestors = 377172605)